pkg/controller/pools_pull_secret_provider/pools_pull_secret_provider.go

```diff
@@ -0,0 +1,137 @@
+		Reconciler: r,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	if err := c.Watch(
+		source.NewKindWithCache(&hivev1.ClusterPool{}, hiveManager.GetCache()),
+		clusterPoolHandler(sourcePullSecretNamespace),
+	); err != nil {
+		return fmt.Errorf("failed to create watch for testimagestreamtagimports: %w", err)
+	}
+
+	r.log.Info("Successfully added reconciler to manager")
+	return nil
+}
+
+func clusterPoolHandler(sourcePullSecretNamespace string) handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(o ctrlruntimeclient.Object) []reconcile.Request {
+		clusterPool, ok := o.(*hivev1.ClusterPool)
+		if !ok {
+			logrus.WithField("type", fmt.Sprintf("%T", o)).Error("Got object that was not an ClusterPool")
+			return nil
+		}
+		if clusterPool.Namespace == sourcePullSecretNamespace {
+			return nil
+		}
+		return []reconcile.Request{{NamespacedName: types.NamespacedName{
+			Namespace: clusterPool.Namespace,
+			Name:      clusterPool.Name,
+		}}}
+	})
+}
+
+type reconciler struct {
+	log                       *logrus.Entry
+	client                    ctrlruntimeclient.Client
+	sourcePullSecretNamespace string
+	sourcePullSecretName      string
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := r.log.WithField("request", req.String())
+	err := r.reconcile(ctx, req, log)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		log.WithError(err).Error("Reconciliation failed")
+	} else {
+		log.Info("Finished reconciliation")
+	}
+	return reconcile.Result{}, controllerutil.SwallowIfTerminal(err)
+}
+
+func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *logrus.Entry) error {
+
+	*log = *log.WithField("namespace", req.Namespace).WithField("name", req.Name)
+	log.Info("Starting reconciliation")
+
+	clusterPool := &hivev1.ClusterPool{}
+	if err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: req.Namespace, Name: req.Name}, clusterPool); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("The cluster pool is deleted")
+			return nil
+		}
+		return fmt.Errorf("failed to get the cluster pool %s in namespace %s: %w", req.Name, req.Namespace, err)
+	}
+
+	if clusterPool.Spec.PullSecretRef == nil {
+		log.Info("The cluster pool does not refer a secret")
+		return nil
+	}
+
+	if clusterPool.Spec.PullSecretRef.Name != r.sourcePullSecretName {
+		log.WithField("clusterPool.Spec.PullSecretRef.Name", clusterPool.Spec.PullSecretRef.Name).
+			Infof("The cluster pool does not refer secret %s", r.sourcePullSecretName)
+		return nil
+	}
+
+	secret := &corev1.Secret{}
+	if err := r.client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: r.sourcePullSecretNamespace, Name: r.sourcePullSecretName}, secret); err != nil {
+		return fmt.Errorf("failed to get the secret %s in namespace %s: %w", r.sourcePullSecretName, r.sourcePullSecretNamespace, err)
+	}
+
+	target := secret.DeepCopy()
```

Comment:
Sorry. I do not understand your comment.
It is copied from the source secret, but the target secret.
Maybe you meant something else?

Reply:
The reason for doing a `DeepCopy` is if you need to keep the unmodified original, for example because you do not want to manipulate your cache or because you want to construct a Patch. You don't do any of that here though, so this is useless, you can just modify and use `secret`

Reply:
Updated.